contracts/staking/src/lib.rs

```diff
@@ -1,4 +1,6 @@
 pub mod contract;
+#[cfg(test)]
+mod contract_test;
```

Comment:
I did this in my early Rust days. It was a result of not understanding Rust well and not realizing the benefits of in-file unit tests (you can test private interfaces ☀️). I'd not do that again and think we should follow Rust best practice.

Reply:
Ok, I only test the public interfaces here and it was getting to be a very large file. I can revert this to just the clippy changes if you prefer.

I thought I finally saw the light of your separate test files

Reply:
I'd prefer to remove c552a60 and just do the clippy fix.

Then the file gets too large, it probably indicates that some functionality plus its tests should be extracted

Reply:
Okay, reverted that commit and added ffcb7ec